### PR TITLE
Use GNUInstallDirs for installation paths

### PR DESCRIPTION
**Pull request by h3xx**
Replace path-guessing system with GNUInstallDirs which provides override-able sane defaults.  To override, pass them in, e.g. `-DCMAKE_INSTALL_LIBDIR=foo`.

Also use these variables for .pc substitutions. `CMAKE_INSTALL_FULL_*` being the full path equivalent to the prefixed version without `FULL`.

This removes the old options.

More information about GNUInstallDirs: https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

### CODE REVIEW

File: libinstpatch-1.0.pc.in [hunk 1]
```diff
@@ -1,7 +1,6 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
```
Comment (derselbst) [inline libinstpatch-1.0.pc.in:2]: Did you intentionally remove the `exec_prefix`? Or should we better keep `exec_prefix=${prefix}` hardcoded in the .pc file?

**Review by derselbst — COMMENTED** [review]
LGTM in general, one question below.

**Comment by h3xx** [comment]
I left it out because YAGNI. I couldn't find anywhere it was depending on that variable. Perhaps you can think of a place it reads that variable.

`pkg-config --variable=exec_prefix libinstpatch-1.0` is the only supported way of reading the variable. It now returns an empty string instead of "/usr" and perhaps this is a breaking change. Not sure what if anything runs that command, though.

Sure, I'll just hard code it to fill it in.
-- 
"If I am to be judged by those who come after me, let me be judged for the truth." — Raistlin Majere

**Review by derselbst — APPROVED** [review]
> `pkg-config --variable=exec_prefix libinstpatch-1.0` is the only supported way of reading the variable.

I don't know how downstream applications are using this. Or if they are using it at all. So better keep it as it was. Thanks!

**Pull request merged**